jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/VirtualThreads.java

```diff
@@ -117,6 +118,29 @@ public static boolean isVirtualThread()
         }
     }
 
+    /**
+     * Get a virtual threads {@code Executor} that names the virtual threads according to the provided name prefix.
+     * While named virtual threads enable observability they do also incur a minor performance penalty.
+     *
+     * @param namePrefix the prefix to use for the name of the virtual threads
+     * @return a virtual threads {@code Executor} that will name the virtual threads according to the provided name prefix.
+     */
+    public static Executor getNamedVirtualThreadsExecutor(String namePrefix)
+    {
+        try
+        {
+            Class<?> builderClass = Class.forName("java.lang.Thread$Builder");
+            Object threadBuilder = Thread.class.getMethod("ofVirtual").invoke(null);
+            threadBuilder = builderClass.getMethod("name", String.class, long.class).invoke(threadBuilder, namePrefix, 0L);
+            ThreadFactory factory = (ThreadFactory)builderClass.getMethod("factory").invoke(threadBuilder);
+            return (Executor)Executors.class.getMethod("newThreadPerTaskExecutor", ThreadFactory.class).invoke(null, factory);
+        }
+        catch (Throwable x)
+        {
+            return null;
```

Comment:
To be consistent with the existing API style (e.g. `getDefaultVirtualThreadsExecutor`), I return `null` here instead of propagating the exception.
Since `org.eclipse.jetty.util.VirtualThreads.Configurable#setVirtualThreadsExecutor` checks for virtual thread support as well as `executor` (and throws an `UnsupportedOperationException` otherwise), we should be fine here.